boards/wio_terminal/src/wifi.rs

```diff
@@ -1,99 +1,106 @@
 pub struct WifiPins {
-    pub pwr: Pa18<Input<Floating>>,
-    pub rxd: Pc22<Input<Floating>>,
-    pub txd: Pc23<Input<Floating>>,
-    pub mosi: Pb24<Input<Floating>>,
-    pub clk: Pb25<Input<Floating>>,
-    pub miso: Pc24<Input<Floating>>,
-    pub cs: Pc25<Input<Floating>>,
-    pub ready: Pc20<Input<Floating>>,
-    pub dir: Pa19<Input<Floating>>,
+    pub pwr: WifiPwrReset,
+    pub rxd: WifiRxdReset,
+    pub txd: WifiTxdReset,
+    pub mosi: WifiTxReset,
+    pub clk: WifiClkReset,
+    pub miso: WifiRxReset,
+    pub cs: WifiCsReset,
+    pub ready: WifiReadyReset,
+    pub dir: WifiDirReset,
 }
 
 /// eRPC-based protocol to the RTL8720 chip
 pub struct Wifi {
-    _pwr: Pa18<Output<PushPull>>,
-    uart: WifiUART,
+    _pwr: WifiPwr,
+    uart: WifiUart,
 
-    rx_buff_isr: Producer<'static, U512>,
-    rx_buff_input: Consumer<'static, U512>,
-    tx_buff_isr: Consumer<'static, U128>,
-    tx_buff_input: Producer<'static, U128>,
+    rx_buff_isr: Producer<'static, 512>,
+    rx_buff_input: Consumer<'static, 512>,
+    tx_buff_isr: Consumer<'static, 128>,
+    tx_buff_input: Producer<'static, 128>,
 
     sequence: u32,
 }
 
-type WifiUART = UART0<Sercom0Pad2<Pc24<PfC>>, Sercom0Pad0<Pb24<PfC>>, (), ()>;
+/// UART pads for the labelled RX & TX pins
+pub type WifiUartPads = uart::Pads<Sercom0, IoSet2, WifiRx, WifiTx>;
+
+/// UART device for the labelled RX & TX pins
+pub type WifiUart = uart::Uart<uart::Config<WifiUartPads>, uart::Duplex>;
 
 impl Wifi {
     pub fn init(
         pins: WifiPins,
         sercom0: SERCOM0,
         clocks: &mut GenericClockController,
         mclk: &mut MCLK,
-        port: &mut Port,
         delay: &mut Delay,
-        rx_buff: &'static BBBuffer<U512>,
-        tx_buff: &'static BBBuffer<U128>,
-    ) -> Result<Wifi, ()> {
+        rx_buff: &'static BBBuffer<512>,
+        tx_buff: &'static BBBuffer<128>,
+    ) -> Wifi {
         let gclk0 = clocks.gclk0();
-        let tx: Sercom0Pad0<_> = pins.mosi.into_pad(port);
-        let rx: Sercom0Pad2<_> = pins.miso.into_pad(port);
-        let uart = UART0::new(
-            &clocks.sercom0_core(&gclk0).ok_or(())?,
-            Hertz(WIFI_UART_BAUD),
-            sercom0,
+
+        let pads = uart::Pads::default().rx(pins.miso).tx(pins.mosi);
+        let uart = uart::Config::new(
             mclk,
-            (rx, tx),
-        );
+            sercom0,
+            pads,
+            clocks.sercom2_core(&gclk0).unwrap().freq(),
+        )
+        .baud(
+            WIFI_UART_BAUD.hz(),
+            uart::BaudMode::Fractional(uart::Oversampling::Bits16),
+        )
+        .enable();
+
         delay.delay_ms(10u8);
 
         // Reset the RTL8720 MCU.
-        let mut pwr = pins.pwr.into_push_pull_output(port);
-        pwr.set_low()?;
+        let mut pwr: WifiPwr = pins.pwr.into();
+        pwr.set_low().ok();
         delay.delay_ms(100u8);
-        pwr.set_high()?;
+        pwr.set_high().ok();
```

Comment:
Same Q as earlier

Reply:
In this instance, @jbeaurivage changed the return type, so `?` is not possible. It looks like `.unwrap()` is used everywhere else in this function (presumably for things that won't actually fail), so I don't see the need to return a `Result` just for setting the pins.